source/array_list.c

```diff
@@ -167,6 +171,7 @@ int aws_array_list_shrink_to_fit(struct aws_array_list *list) {
 
 int aws_array_list_copy(const struct aws_array_list *from, struct aws_array_list *to) {
     assert(from->item_size == to->item_size);
+    assert(from->data);
```

Comment:
We can handle it if you modify that if-statements that follow, like so:
```
    ...
    if (to->current_size >= copy_size) {
        if (copy_size > 0) {
            memcpy(to->data, from->data, copy_size);
        }
        ...
```
This is also more efficient than the current code, which needlessly re-allocates when `to->len` is less than capacity

Reply:
good idea. Done.